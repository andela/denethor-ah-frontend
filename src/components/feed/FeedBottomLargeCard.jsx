import React from 'react';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faClock } from '@fortawesome/free-solid-svg-icons';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import './styles.scss';
import validator from 'validator';

library.add(faClock);

const FeedBottomLargeCard = ({
  id,
  title,
  description,
  dateCreated,
  author = {},
  featuredImage,
  readTime
}) => {
  title = validator.unescape(title);
  return (
    <div className="feed-bottom-large-card-container">   
        <div className="feed-bottom-large-card-container-image">
          <img alt="feed" src={ featuredImage } />
          <div className='feed-bottom-large-card-container-overlay' />
        </div>
        <div>
          <div className="feed-bottom-container__large_image_info-container">
            <h3 className="feed-bottom-container__large_image__info-header"><NavLink to={`/articles/${id}`}>{title}</NavLink></h3>
            <p className="feed-bottom-container__large_image__info-published-date">{author.username}</p>
            <p className="feed-bottom-container__large_image__info-minutes-read">
              {dateCreated} &nbsp;&nbsp;&nbsp;
              <FontAwesomeIcon icon="clock" color="#818181" size="1x"/>    
              {`  ${readTime} `}
            </p>
          </div>
        </div> 
        <div className="feed-bottom-container__article-description">
          {description}
          </div>    
    </div>
  )
}

FeedBottomLargeCard.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string,
  dateCreated: PropTypes.string,
  featuredImage: PropTypes.string,
  body: PropTypes.string,
  createdAt: PropTypes.string,
  author: PropTypes.object,
  readTime: PropTypes.string
}

export default FeedBottomLargeCard;