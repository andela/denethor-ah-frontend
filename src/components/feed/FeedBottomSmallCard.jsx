import React from 'react';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faClock } from '@fortawesome/free-solid-svg-icons';
import { monthDifference } from '../../utils/monthsDifference';
import { extractImageFromBody } from '../../utils/imageExtractor';
import PropTypes from 'prop-types';
import './styles.scss';

library.add(faClock);

const FeedBottomSmallCard = ({
  title,
  body,
  createdAt,
  author,
  readTime
}) => {

  const articleImage = extractImageFromBody(body);
  return (
    <div className="feed-bottom-container-small">   
        <img src = {articleImage} alt="feed" className="feed-bottom-container__small_image"/> 
        <div className="feed-bottom-container__small_image_info-container">
            <h3 className="feed-bottom-container__small_image__info-header">{title}</h3>
            <p className="feed-bottom-container__small_image__info-published-date">{`${author.username} ${monthDifference(createdAt)} months ago`}</p>
            <p className="feed-bottom-container__small_image__info-minutes-read">
              <FontAwesomeIcon icon="clock" color="#818181" size="1x"/>    
              {` ${readTime}`}
            </p>
          </div>
    </div>
  )
}

FeedBottomSmallCard.propTypes = {
  title: PropTypes.string,
  body: PropTypes.string,
  createdAt: PropTypes.string,
  author: PropTypes.object,
  readTime: PropTypes.string
}

export default FeedBottomSmallCard;