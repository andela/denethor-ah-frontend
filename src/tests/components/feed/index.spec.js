import React from 'react';
import { mount } from 'enzyme';
import FeedBottomLargeCard from '../../../components/feed/FeedBottomLargeCard';
import FeedBottomSmallCard from '../../../components/feed/FeedBottomSmallCard';

describe('Test for the feed bottom component', () => {
  const body = "<strong>Our original read time calculation was geared toward “slow” images, like comics, where you would really want to sit down and invest in the image. </strong>\n\nThis resulted in articles with crazy big read times. For instance, this article containing 140 images was clocking in at a whopping 87 minute read. So we amended our read time calculation to count 12 seconds for the first image, 11 for the second, and minus an additional second for each subsequent image. Any images after the tenth image are counted at three seconds.\n\n<img src='https://res.cloudinary.com/jsamcloud12/image/upload/v1552053108/blur-charge-close-up-256302.jpg' alt='featured' />\n\n    You might see this change reflected across the site. Keep in mind that our estimated read time is just that: an estimation. You might finish a story faster or slower depending on various factors such as how many children or cats you have, your caffeine/alcohol intake, or if you’re a time-traveler from the future and already read that story. We just want to give you a ballpark figure so you can decide whether you have time to read one more story before the bus comes, or if you should bookmark it for later.\n\n<img src='https://images.unsplash.com/photo-1551632436-cbf8dd35adfa?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=751&q=80' alt='featured' />\n\n\nYou begin with a text, you sculpt information, you chisel away what's not needed, you come to the point, make things clear, add value, you're a content person, you like words. Design is no afterthought, far from it, but it comes in a deserved second. Anyway, you still use Lorem Ipsum and rightly so, as it will always have a place in the web workers toolbox, as things happen, not always the way you like it, not always in the preferred order. Even if your less into design and more into content strategy you may find some redeeming value with, wait for it, dummy copy, no less.\n\n\nSo Lorem Ipsum is bad (not necessarily)\nI’ve heard the argument that “lorem ipsum” is effective in wireframing or design because it helps people focus on the actual layout, or color scheme, or whatever. What kills me here is that we’re talking about creating a user experience that will (whether we like it or not) be DRIVEN by words. The entire structure of the page or app flow is FOR THE WORDS.\nIf that's what you think how bout the other way around? How can you evaluate content with"
  const author = { username: 'Princess'}  
  const createdAt = '2018-03-10';
  it('should contain FeedBottomLargeCard ', () => {
        const wrapper = mount( < FeedBottomLargeCard  body={ body} author ={author } createdAt= {createdAt} /> );
        expect(wrapper.length).toBe(1);
        expect(wrapper.find('div').length).toBe(5);
        expect(wrapper.find('h3').length).toBe(1);
        expect(wrapper.find('p').length).toBe(2);
    });
    it('should contain FeedBottomSmallCard correctly', () => {
      const wrapper = mount( < FeedBottomSmallCard body = { body } author ={author } createdAt= {createdAt} /> );
      expect(wrapper.length).toBe(1);
      expect(wrapper.find('div').length).toBe(2);
      expect(wrapper.find('h3').length).toBe(1);
      expect(wrapper.find('p').length).toBe(2);
  })
})

