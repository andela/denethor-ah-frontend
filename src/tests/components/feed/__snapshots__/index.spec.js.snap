// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the feed bottom component should fail to dislike an article 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
